test/integration/openapiconv/convert_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapiconv
+
+import (
+	"log"
+	"reflect"
+	"testing"
+
+	builderv2 "k8s.io/kube-openapi/pkg/builder"
+	builderv3 "k8s.io/kube-openapi/pkg/builder3"
+	"k8s.io/kube-openapi/pkg/openapiconv"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"k8s.io/kube-openapi/test/integration/pkg/generated"
+	"k8s.io/kube-openapi/test/integration/testutil"
+)
+
+func TestConvertGolden(t *testing.T) {
```

Comment:
Can we update this test (and the other tests) to make sure that the conversion function DOES NOT alter the openapi v2?

Reply:
The conversion function does alter the openapi. I want to point out that the aggregator would download the OpenAPI V2 json, unmarshal, runs the V2->V3 converter, and then directly marshal the V3 to be served. The intermediary OpenAPI v2 spec is not used for anything else.

To ensure that the conversion function does not alter the openapi v2, we need to deepcopy the openapi v2. That would result in a marshal/unmarshal in the converter. The aggregator would then marshal/unmarshal the openapi v2 multiple times to preserve our unaltered openapi v2 guarantee.

One way I can think of doing this is changing the function signature to accept the marshalled swagger instead of the swagger itself. That will save us on a json round trip if we want to guarantee that v2 schema is unaltered.

Reply:
That's awesome thanks